File: DesignPatternsExercises/java/AbstractFactoryDesignPattern/Solution/PostCodeFactory.java
public class PostCodeFactory implements AbstractFactory {

	@Override
	public PostCode getPostCodeInstance(Country country, String postalCode) {
		if (country == Country.US)
			// return new __________(13)__________(postalCode);
			return new USPostCode(postalCode);
		if (country == Country.UK)
			// return new __________(14)__________(postalCode);
			return new UKPostCode(postalCode);
		return null;
	}

	@Override
	public PhoneNumber getPhoneNumberInstance(Country country, String phoneNumber) {
		throw new UnsupportedOperationException("Unable to get PhoneNumberInstance in the PostCodeFactory");
	}
}

File: DesignPatternsExercises/ruby/abstract_factory_design_pattern/solution/post_code_factory.rb
require_relative 'us_post_code'
require_relative 'uk_post_code'

class PostCodeFactory
  # def __________(11)__________(country, post_code)
  def instance(country, post_code)
    # return __________(12)__________(post_code) if country == :US
    return USPostCode.new(post_code) if country == :US
    # return __________(13)__________(post_code) if country == :UK
    return UKPostCode.new(post_code) if country == :UK

    raise 'Country is not supported'
  end
end
